### PR TITLE
Update reference.rst 

### PR DESCRIPTION
**Pull request by strawhat54**
Fixed contradiction for  [issue #12965](https://github.com/ipython/ipython/issues/12965)

### CODE REVIEW

File: IPython/core/magics/code.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def pastebin(self, parameter_s=''):
         ).encode("utf-8")
 
         request = Request(
-            "http://dpaste.com/api/v2/",
+            "https://dpaste.com/api/v2/",
```
Comment (MrMino) [inline IPython/core/magics/code.py:279]: That should be a separate PR :). Please don't mix these two things.

Comment (strawhat54) [reply IPython/core/magics/code.py:279]: Yes, very sorry. I thought that i cancelled the second PR. I will do so right away!

**Comment by MrMino** [comment]
@strawhat54 I think you reverted the wrong commit. The `https` thing is not related to #12965.

No biggie, just bring back the changes related to documentation here and revert cf06651.

I know it might seem like I'm splitting hairs here, but I need to ensure that you, as a first-time contributor, understand how PRs work :).

**Comment by strawhat54** [comment]
Hey @MrMino , Just wanted to thank you for your co operation :heart:  . I thought that I messed up my first PR and gonna get a real scolding but, instead, you helped me till the end and made sure I understood everything. I am really grateful to you and will do my best to contribute to open source from now on :)

**Comment by MrMino** [comment]
Thanks @strawhat54! I'm glad to hear that!
